registry/storage/driver/s3/s3.go

```diff
@@ -587,6 +587,11 @@ func (d *driver) List(path string) ([]string, error) {
 	if path != "/" && path[len(path)-1] != '/' {
 		path = path + "/"
 	}
+	prefix := ""
```

Comment:
This is the only call that does that. If there were more, I would have modularized it. It also was fairly easy until now :). I can think of a way to generalize it, but I think the cases for files and directories are somewhat different.